ttnn/cpp/ttnn/operations/reduction/generic/device/reduce_op.cpp

```diff
@@ -77,24 +108,27 @@ std::vector<tt::tt_metal::LegacyShape> Reduce::compute_output_shapes(const std::
             padding[3] = Padding::PadDimension{0, 31};
             break;
     }
-    return {tt::tt_metal::LegacyShape(output_shape, padding)};
+    auto output_shape_with_padding = ttnn::Shape(tt::tt_metal::LegacyShape(output_shape, padding));
+    return {std::pair(output_shape_with_padding.logical_shape(), TensorLayout::fromLegacyPaddedShape(this->output_dtype, PageConfig(Layout::TILE), this->output_mem_config, output_shape_with_padding.padded_shape()))};
 }
```

Comment:
Need to properly figure out the output tensor layout without relying on current logic with `LegacyShape` and padding.